### PR TITLE
M365DSCReport: Add support for creating a report in CSV-format

### PR DESCRIPTION
**Pull request by salbeck-sit**
#### Pull Request (PR) description
This PR adds support for -Type CSV in New-M365DSCReportFromConfiguration and adds dynamic parameter -Delimiter when -Type CSV is used

#### This Pull Request (PR) fixes the following issues
I'm having issues creating an Excel-report of a configuration. On my local PC where I'm not a local admin the generation of the report fails with a CIM-error and on my admin-server I don't have Excel available. The obvious solution (for me) is to create the report in CSV-format and import it into Excel later.

#### Task list

- [x] Added an entry to the change log under the Unreleased section of the file CHANGELOG.md.
      Entry should say what was changed and how that affects users (if applicable), and
      reference the issue being resolved (if applicable).
- [ ] Resource parameter descriptions added/updated in the schema.mof.
- [ ] Resource documentation added/updated in README.md.
- [ ] Resource settings.json file contains all required permissions.
- [ ] Examples appropriately added/updated.
- [ ] Unit tests added/updated.
- [x] New/changed code adheres to [DSC Community Style Guidelines](https://dsccommunity.org/styleguidelines).

### CODE REVIEW

File: Modules/Microsoft365DSC/Modules/M365DSCReport.psm1 [hunk 1]
```diff
@@ -554,6 +554,107 @@ function New-M365DSCConfigurationToExcel
...
+    $modelRow = @{'Component Name'=$null; Property=$null; Value = $null}
+    $row = 0
+    $csvOutput = @()
+
+    foreach ($resource in $parsedContent)
+    {
+        $newRow = $modelRow.Clone()
+        if ($row -gt 0)
+        {
+            write-verbose "add separator-line in CSV-file between resources"
```
Comment (FabienTschanz) [inline Modules/Microsoft365DSC/Modules/M365DSCReport.psm1]:
```suggestion
            Write-Verbose -Message "Add separator-line in CSV-file between resources"
```

File: Modules/Microsoft365DSC/Modules/M365DSCReport.psm1 [hunk 2]
```diff
@@ -554,6 +554,107 @@ function New-M365DSCConfigurationToExcel
...
+                        else
+                        {
+                            $value = ($resource.$property).ToString().Replace('$', '')
+                            $value = $value.Replace('@', '')
+                            $value = $value.Replace('(', '')
+                            $value = $value.Replace(')', '')
+                            $newRow.Value = $value
+                        }
+                    }
+
```
Comment (FabienTschanz) [inline Modules/Microsoft365DSC/Modules/M365DSCReport.psm1]: Please remove this empty line.

File: Modules/Microsoft365DSC/Modules/M365DSCReport.psm1 [hunk 3]
```diff
@@ -598,6 +699,31 @@ function New-M365DSCReportFromConfiguration
...
+begin
+{
+    if ($PSBoundParameters.ContainsKey('Delimiter'))
+    {
+        $Delimiter = $PSBoundParameters.Delimiter
+    }
+}
+process # required with DynamicParam
+{
```
Comment (FabienTschanz) [inline Modules/Microsoft365DSC/Modules/M365DSCReport.psm1]: Indentation is not correct, leading to a curly bracket on the same line as another end block on line 778.

**Review by ykuijs — APPROVED** [review]
LGTM

**Pull request merged**